src/WebJobs.Script.WebHost/Models/RunFromPackageContext.cs

```diff
@@ -32,29 +35,10 @@ public bool IsScmRunFromPackage()
                         StringComparison.OrdinalIgnoreCase);
         }
 
-        public async Task<bool> BlobExistsAsync(ILogger logger)
+        public async Task<bool> IsRunFromPackage(ILogger logger)
         {
-            bool exists = false;
-            await Utility.InvokeWithRetriesAsync(async () =>
-            {
-                try
-                {
-                    CloudBlockBlob blob = new CloudBlockBlob(new Uri(Url));
-                    exists = await blob.ExistsAsync();
-                }
-                catch (Exception e)
-                {
-                    logger.LogError(e, $"Failed to check if zip url blob exists");
-                    throw;
-                }
-            }, maxRetries: 2, retryInterval: TimeSpan.FromSeconds(0.3));
-
-            if (!exists)
-            {
-                logger.LogWarning($"{EnvironmentVariableName} points to an empty location. Function app has no content.");
-            }
-
-            return exists;
+            return (IsScmRunFromPackage() && await _runFromPackageCloudBlockBlobService.BlobExists(Url, EnvironmentVariableName, logger)) ||
```

Comment:
Apologies for piling on this PR - with this check in place - can we address issue @pgombar was running into with scm run from package  for isReadOnlyFileSystem check?

Reply:
Yes, We could. But I think better/easier to make that change as a separate PR.